File: garage_server/model/movie.go
package model

import (
	"time"
)

type Movie struct {
	Id             uint       `json:"id"`
	Code           string     `json:"code" validate:"required"`
	Title          string     `json:"title" validate:"required"`
	Cover          *string    `json:"cover"`
	PublishDate    *time.Time `json:"publish_date"`
	Director       *string    `json:"director"`
	ProduceCompany *string    `json:"produce_company"`
	PublishCompany *string    `json:"publish_company"`
	Series         *string    `json:"series"`
	CreatedAt      *time.Time `json:"created_at"`
	UpdatedAt      *time.Time `json:"updated_at"`
}

type MovieInfo struct {
	Movie  Movie        `json:"movie"`
	Actors []MovieActor `json:"actors"`
	Tags   []MovieTag   `json:"tags"`
}

File: garage_web/src/api/movie.ts
import { http, Response } from "@/utils/http";
import { MovieTag } from "./movie_tag";
import { MovieActor } from "./movie_actor";

export interface Movie {
  id: number;
  code: string;
  title: string;
  cover?: string | null;
  publishDate?: Date | null;
  director?: string | null;
  produceCompany?: string | null;
  publishCompany?: string | null;
  series?: string | null;
  createdAt?: Date | null;
  updatedAt?: Date | null;
}

export interface MovieInfo {
  movie: Movie;
  actors: MovieActor[];
  tags: MovieTag[];
}

export const GetMovies = () => {
  return http.get<Response<Array<Movie>>>("/movie");
};

export const GetMovieInfo = (code: string) => {
  return http.get<Response<MovieInfo>>(`/movie/info/${code}`);
};

export const CreateMovies = () => {
  return http.post<Response<null>>("/movie");
};
